### DISCUSSION TITLE
Will this work with a R5 + SEC module?

### DISCUSSION THREAD

**Discussion by santiagoGPNC** [paolosabatino/saj-mqtt-ha#11] [Q&A]
Hi! I have a R5 inverter (rebranded as greenheiss) with a SEC Module. Does anyone knows if this integration would be compatible with my setup?
the SEC module also pushes mqtt messages to saj, so it should be rather similar, but maybe the messages are different. Im not certain.

Any info would be appreciated.

**Comment by santiagoGPNC** [comment]
a bit more info, in case it is useful:
-I have poisoned the DNS and I see messages for topics _saj/serialnumber/realtime_data_ and _saj/serialnumber/data_transmission_, but NOT data_transmission_rsp.
-I also see some homeassistant specific topics, so that connection seem to be working.
-I get an error in the logs: the inverter did not answer in the expected timeout.

Im guessing that while both inverters use mqtt, their mechanism might not be the same, but Im just starting learning this and there's too much I dont know.

**Comment by paolosabatino** [comment]
hello; the integration has never been tested with anything else than H1. The protocol for other models may be slightly different or heavily different, but since there is no public documentation it is impossible to say anything about.

By the way, I suggest you to find/ask the proper documentation and try the other saj-mqtt project which allows manual queries to the inverter via mqtt to see the direct answers; the modbus registers documentation is essential though:

https://github.com/paolosabatino/saj-mqtt

**Reply by santiagoGPNC** [reply]
thank you Paolo.
I know next to zero about modbus so this has been challenging for me. I've been looking for information specific to the SEC Module but I cannot find anything specific about it.
Do you know if SAJ itself is open to give hints on the matteR?
